colorama/ansitowin32.py

```diff
@@ -80,12 +88,41 @@ def __init__(self, wrapped, convert=None, strip=None, autoreset=False):
         # create the proxy wrapping our output stream
         self.stream = StreamWrapper(wrapped, self)
 
+        conversion_supported = True
         on_windows = os.name == 'nt'
         # We test if the WinAPI works, because even if we are on Windows
         # we may be using a terminal that doesn't support the WinAPI
         # (e.g. Cygwin Terminal). In this case it's up to the terminal
         # to support the ANSI codes.
-        conversion_supported = on_windows and winapi_test()
+        if on_windows and platform.release().startswith('10'):
+            # Windows 10 supports ANSI sequences using virtual terminal mode.
+            # However, this has been turned off by default from an early Windows 10 version.
+            # We turn virtual terminal support back on, then set
+            # `conversion_supported` to False. Behaviour will then be the same
+            # as for Linux - colorama doesn't use win32 calls.
+            from ctypes import wintypes, byref  # import wintypes now we know we're on Windows.
+            old_mode = wintypes.DWORD()
+
+            stdout_handle = windll.kernel32.GetStdHandle(STDOUT)
+            # If the next call failed, we may be running on Windows pre 10.0.10586.
```

Comment:
If I understand correctly, this call isn't supposed to fail (`GetConsoleMode` existed in prior versions), but it will not return the `ENABLE_VIRTUAL_TERMINAL_PROCESSING` flag which only exists on later versions. The "set" call with the flag may fail, though. Please update the comments to reflect this.

Reply:
Actually, I think there's more than just the comment to change. The only way to be sure that we've enabled the virtual terminal is to call GetConsoleMode again, and check. Only if that test passes should I set conversion_supported to False.
To be safe, I need to find an old version of Windows 10 to test on. I'll also test on Cygwin.